app/models/event_schedule.rb

```diff
@@ -0,0 +1,27 @@
+class EventSchedule < ActiveRecord::Base
+  belongs_to :schedule
+  belongs_to :event
+  belongs_to :room
```

Comment:
Maybe you should add some validations for schedule, room and event. Then you don't need the .try everywhere and I'm not sure if an EventSchedule without these associations makes sense!